test/libyul/yulOptimizerTests/loadResolver/mload_in_function.yul

```diff
@@ -0,0 +1,29 @@
+{
+        function foo(x) {
+                for {} x { x := mload(0) mstore(0, 0)} {}
+        }
+        mstore(0, 1337)
+        foo(42)
+        sstore(0, mload(0))
+}
+// ====
+// step: loadResolver
+// ----
+// {
+//     function foo(x)
+//     {
+//         for { }
+//         x
+//         {
+//             let _1 := 0
+//             x := mload(_1)
+//             mstore(_1, _1)
+//         }
+//         { }
+//     }
+//     let _4 := 1337
+//     let _5 := 0
+//     mstore(_5, _4)
+//     foo(42)
+//     sstore(_5, mload(_5))
+// }
```

Comment:
Looks good. Could you please add the following test cases in `loadResolver` as well (the expectation may need to be adapted because of whitespace mismatch)

```
{
    function userNot(x) -> y {
        y := iszero(x)
    }

    function funcWithLoop(x) {
        for {} userNot(x) { mstore(0, 0) } {}
    }

    mstore(0, 1337)
    funcWithLoop(42)
    sstore(0, mload(0))
}
// ====
// step: loadResolver
// ----
// { 
//     function userNot(x) -> y
//     { y := iszero(x) }
//     function funcWithLoop(x_1)
//     {
//         let _1 := 0
//         mstore(_1, _1)
//         for { } userNot(x_1) { }
//         { }
//     }
//     let _3 := 1337
//     let _4 := 0
//     mstore(_4, _3)
//     funcWithLoop(42)
//     sstore(_4, mload(_4))
// }
```

and 

```
{
    function userNot(x) -> y {
        y := iszero(x)
    }

    function funcWithLoop(x) {
        for { mstore(0, 0) } userNot(x) {} {}
    }

    mstore(0, 1337)
    funcWithLoop(42)
    sstore(0, mload(0))
}
// ====
// step: loadResolver
// ----
// { 
//     function userNot(x) -> y
//     { y := iszero(x) }
//     function funcWithLoop(x_1)
//     {
//         let _1 := 0
//         mstore(_1, _1)
//         for { } userNot(x_1) { }
//         { }
//     }
//     let _3 := 1337
//     let _4 := 0
//     mstore(_4, _3)
//     funcWithLoop(42)
//     sstore(_4, mload(_4))
// }
```